app/controllers/task_controller.rb

```diff
@@ -17,24 +19,69 @@ def new
   end
 
   def create
-    @task = Task.new(create_params[:task])
-    @task.save
+     # Guard clause against empty tasknames
+    @taskname = params[:task][:taskname]
+     #Checks if string is empty, sets it to null to trigger db-level error
+    if @taskname.length == 0
+      @taskname = nil
+    else
+      @task = Task.new(create_params[:task])
+    end
 
+    @task.save
     redirect_to root_url # This is preferred if the root is not '/'
+
+    # What the user sees if they his a db-level error
+    rescue => error
+      render :fancy_error
+
   end
 
   def destroy
     # local variables here cuz it's not being displayed / used elsewhere
     id = params[:id]
-    onetask = Task.find(id).destroy
+    destroytask = Task.find(id).destroy
+
+    redirect_to root_url
+  end
+
+  def update_status
+    id = params[:id]
+    update = Task.find(id)
+
+    # Allows you to toggle the read status between read and unread
+    if update.comp_status == "unread"
```

Comment:
If you only have a "unread" vs "read" status - you could potentially just use a boolean in your database rather than a string.